pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -437,9 +613,31 @@ func generatePeerConfigs(subnets []string, gatewayIP net.IP) (map[*net.IPNet]net
 }
 
 // getPeerGatewayIP will always return the first Gateway IP.
-func getPeerGatewayIP(spec mcv1alpha1.ClusterInfo) net.IP {
+func getPeerGatewayIP(spec mcv1alpha1.ClusterInfo, enableWireGuard bool) net.IP {
+	if enableWireGuard {
+		_, serviceCIDR, err := net.ParseCIDR(spec.ServiceCIDR)
+		if err != nil {
+			return nil
+		}
+		return serviceCIDR.IP.Mask(serviceCIDR.Mask)
+	}
 	if len(spec.GatewayInfos) == 0 {
 		return nil
 	}
 	return net.ParseIP(spec.GatewayInfos[0].GatewayIP)
 }
+
+func getLocalGatewayIP(gateway *mcv1alpha1.Gateway, serviceCIDRProvider servicecidr.Interface) net.IP {
+	if gateway.WireGuard != nil {
+		serviceCIDR, err := serviceCIDRProvider.GetServiceCIDR(false)
+		if err != nil {
+			return nil
```

Comment:
ditto, better to have a log to capture the error.

Reply:
Add a log.